src/pyodide-py/pyodide/_base.py

```diff
@@ -100,61 +102,25 @@ def eval_code(code: str, ns: Dict[str, Any]) -> None:
     if len(mod.body) == 0:
         return None
 
-    target_name = "<EXEC-LAST-EXPRESSION>"
-    mod = _adjust_ast_to_store_result(target_name, mod, code)
-    eval(compile(mod, "<exec>", mode="exec"), ns, ns)
-    return ns.pop(target_name)
+    # we extract last expression
+    last_expr = None
+    if isinstance(mod.body[-1], (ast.Expr, ast.Await)) and not quiet_code(code):
```

Comment:
You should combine 
`isinstance(mod.body[-1], (ast.Expr, ast.Await))` into `quiet_code`, probably rename `quiet_code` to `should_code_return_value` or something like that. Then we can keep my tests but apply them to `should_code_return_value`.